src/Gateway.php

```diff
@@ -81,6 +82,17 @@ protected function massageSignatureData(array $data)
         }, $data);
     }
 
+    /**
+     * Flattens signature data
+     *
+     * @param array $data
+     * @return array
+     */
+    protected function flattenSignatureData($data)
+    {
+        return iterator_to_array(new \RecursiveIteratorIterator(new \RecursiveArrayIterator($data)), false);
```

Comment:
Why is this required?

Reply:
The data that is used for generating the signature must be a flattened. Before this PR it already was a flattened array. But with the addition of shippingdetail data the $data array becomes multidimensional.

The flattenSignatureData code is copied from the Omnikassa V1 repository. I have tested it with a Omnikassa V2 test account.

Example:
```
// $data
$data['merchantOrderId'] = 'xyz';
$data['shippingdetail'] = [
	'firstname' => 'x',
	'lastname' => 'y',
	'etc ...' => 'etc ...'
];

// Must be converted to this
$signatureData = [
        'merchantOrderId' => 'xyz',
	'firstname' => 'x',
	'lastname' => 'y',
	'etc ...' => 'etc ...'
];
```